app/Helpers/StaffingHelper.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Models\Staffing;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Http;
+
+class StaffingHelper
+{
+    public static function resetStaffing(Staffing $staffing) 
+    {
+        $event = $staffing->event;
+
+        $childEvent = $event->parent()->exists() ? $event->parent->children()->where('start_date', '>', Carbon::now())->first() : $event->children()->where('start_date', '>', Carbon::now())->first();
+
+        if(!$childEvent) {
+            throw new \Exception('No future events found.');
+            return false;
+        }
+
+        $staffing->event()->associate($childEvent);
+        $staffing->positions()->each(function($position) {
+            if ($position->booking_id) {
+                $response = Http::withToken(config('booking.cc_api_token'))->delete(config('booking.cc_api_url') . '/bookings/' . $position->booking_id);
+
+                if ($response->failed()) {
+                    throw new \Exception('Failed to delete booking. Error: ' . $response->body());
+                    return false;
+                }
+
+                $position->booking_id = null;
+            }
+
+            $position->discord_user = null;
+            $position->save();
+        });
+
+        $staffing->save();
+
+        return true;
+    }
+
+    public static function updateDiscordMessage(Staffing $staffing, $reset = null)
+    {
+        $payload = [
+            'id' => $staffing->id
+        ];
+
+        if ($reset) {
+            $payload['reset'] = $reset;
+        }
+
+        // Send a api request to the discord bot to update the staffing message
+        $response = Http::withToken(config('booking.discord_api_token'))->asForm()->post(config('booking.discord_api_url') . '/staffings/update', $payload);
+
+        if ($response->failed()) {
+            throw new \Exception('Failed to update Discord message. Error: ' . $response->body());
+            return false;
+        }
+
+        return $response->json();
+    }
+
+    public static function setupStaffing(Staffing $staffing)
```

Comment:
nitpick non-blocking suggestion: use `setup()` because Staffing is already a parameter.

Likewise for `resetStaffing`, make that `reset()` instead.